spring-cloud-zookeeper-discovery/src/test/groovy/org/springframework/cloud/zookeeper/discovery/base/AvailablePortScanner.groovy

```diff
@@ -0,0 +1,79 @@
+package org.springframework.cloud.zookeeper.discovery.base
+
+import groovy.transform.CompileStatic
+import groovy.transform.PackageScope
+import groovy.util.logging.Slf4j
+import org.apache.commons.lang.math.RandomUtils
+import org.apache.tomcat.util.http.fileupload.IOUtils
+
+@CompileStatic
+@Slf4j
+class AvailablePortScanner {
```

Comment:
Spring has something similar to this: http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/util/SocketUtils.html#findAvailableTcpPort--